src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ServiceDescriptor.cs

```diff
@@ -262,7 +263,17 @@ public override string ToString()
 
                 if (KeyedImplementationFactory != null)
                 {
-                    return lifetime + $"{nameof(KeyedImplementationFactory)}: {KeyedImplementationFactory.Method}";
+#if NET9_0_OR_GREATER
```

Comment:
nit: repo-wide, this one is more prevalent:
```suggestion
#if NET
```

Reply:
I don't prefer writing long ifdefs either but this builds for .NET 8 and the API was only added in 9, so...

https://github.com/dotnet/runtime/blob/d92006acaf3a64670f3a2f4bb3c7b85c9c3c45f8/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/Microsoft.Extensions.DependencyInjection.Abstractions.csproj#L4

Reply:
Ah, this is one published as a nuget package. I forgot about it. Yup, they get cleaned up when NetCoreAppMinimum is bumped, all good.